source/workshop_1_mcmc_en_brms_eng.Rmd

```diff
@@ -0,0 +1,1195 @@
+  # calculate summary statistics for each variable
+  group_by(.variable) %>%
+  summarise(min = min(.value),
+            q_05 = quantile(.value, probs = 0.05),
+            q_20 = quantile(.value, probs = 0.20),
+            gemiddelde = mean(.value),
+            mediaan = median(.value),
+            q_80 = quantile(.value, probs = 0.80),
+            q_95 = quantile(.value, probs = 0.95),
+            max = max(.value))
+```
+
+Useful functions of the **tidybayes** package are also `median_qi()`, `mean_qi()` ... after `gather_draws()` which you can use instead of `group_by()` and `summarise()` .
+
+We would now like to visualize the estimated number of species per habitat type with associated uncertainty. With the function `spread_draws()` you can take a certain number of samples from the posterior distribution and convert them into a wide format table. The average number of species in swamps according to our model is $\exp(\beta_0)$ and in forests $\exp(\beta_0+\beta_1)$. We show the posterior distributions with the posterior median and 60 and 90% credible intervals.
+
+```{r resultats-fit-poisson-3}
+fit_poisson2 %>%
+  # vgather 1000 posterior samples for 2 parameters in wide format
+  spread_draws(b_Intercept, b_habitatForest, ndraws = 1000, seed = 123) %>%
+  # calculate average numbers and convert to long format for visualization
+  mutate(bog = exp(b_Intercept),
+         forest = exp(b_Intercept + b_habitatForest)) %>%
+  pivot_longer(cols = c("bog", "forest"), names_to = "habitat", 
+               values_to = "sp_rich") %>%
+  # visualise via ggplot()
+  ggplot(aes(y = sp_rich, x = habitat)) +
+    stat_eye(point_interval = "median_qi", .width = c(0.6, 0.9)) +
+    scale_y_continuous(limits = c(0, NA))
+```
+
+In addition to `stat_eye()` you will find [here](https://mjskay.github.io/tidybayes/articles/tidy-brms.html#other-visualizations-of-distributions-stat_slabinterval) some nice ways to visualize posterior distributions .
+
+We see a clear difference in the number of species between the two habitats. Is there a significant difference between the number of species in swamps and forests? We test the hypothesis that numbers are equal in swamps and forests.
+
+$$
+\exp(\beta_0) = \exp(\beta_0+\beta_1)\\
+\Rightarrow \beta_0 = \beta_0 + \beta_1\\
+\Rightarrow \beta_1 = 0\\
+$$
+
+This can easily be done via the `hypothesis()` function of the **brms** package.
+The argument `alpha` specifies the size of the credible interval.
+This allows hypothesis testing in a similar way to the frequentist null hypothesis testing framework.
+
+```{r resultats-hypothesis-test}
+# Test hypothesis difference between habitats
+hyp <- hypothesis(fit_poisson2, "habitatForest = 0", alpha = 0.1)
+hyp
+```
+
+```{r resultats-hypothesis-test-vis}
+# Plot posterior distribution hypothesis
+plot(hyp)
+```
+
+Finally, we visualise the random effects of the sites. We sort them from high to low species richness.
```

Comment:
```suggestion
We can conclude that there is a significant difference since 0 is not included in the 90% credible interval.

Finally, we visualise the random effects of the sites. We sort them from high to low species richness.
```